### ISSUE TITLE
Сan I trust CB for strong anonymity?

### ISSUE THREAD

**Issue by ghost** [kkapsner/CanvasBlocker#370]

**Comment by kkapsner** [comment]
It depends on what you exactly mean by "strong" and how the page you are using generates the fingerprint. There different types of entropy in the generated canvas that allow the fingerprinting.
CanvasBlocker protects against the fine differences between the computed images (not visible to the human eye) by adding some noise to it. But bigger differences (e.g. different installed fonts that are printed) are still visible and can be tracked (if the fingerprinter only looks for them) - but this does not allow for individual tracking down to the actual GPU.

As far as I know the most common fingerprints are generated using both types. This helps them to collect as many entropy as possible to have the maximal tracking potential.

So in conclusion: it should not be possible to extract the original fingerprint. But it is possible to calculate some similarity index between the original and the faked one. But you should share the same index with other people.

**Comment by jozer-dev** [comment]
@kkapsner Thank you for your answer! I should've given more detail. You say "you should share the same index with other people" and what i mean by "recover" . For exapme, i have facebook account and facebook read out and know my original canvas fingerprint with defualt browser fonts and installed ms office in system. I've got a situation where I'm having trouble with the law and I need to register my Facebook (I know I can't upload my photo and communicate with friends and acquaintances). If I singup with privacy.resistFingerprinting, Facebook can block me or highlight me. Ok, I installed new system, installed some additional fonts in the system, changed the default fonts in my browser, installed firefox and CB with persistent canvas fingerprint, I have a another IP (public Wi-Fi). What percentage of me will be found with the canvas fingerprint data that Facebook gave?

Thank you very much for support.

**Comment by concatime** [comment]
`privacy.resistFingerprinting` is not bad when combined with CanvasBlocker. In Firefox, you can hide your system fonts with `browser.display.use_document_fonts` set to 0. This is only one in many options to consider.

**Comment by kkapsner** [comment]
The original author mentioned in another post that it was about Facebook. Given the number of Facebook users I think they are not able to track someone individually based on the the bigger differences in the canvas.
